lib/trento.pm

```diff
@@ -826,6 +828,43 @@ sub cluster_wait_status {
     die "Timeout waiting for the change" if !$done;
 }
 
+=head3 cluster_wait_status_by_regex
+
+Remotly run 'SAPHanaSR-showAttr' in a loop on $host, wait output that match using regular expression
+=cut
+
+sub cluster_wait_status_by_regex {
+    my ($host, $timeout, $regular_expression, $text_to_find) = @_;
+    $timeout //= bmwqemu::scale_timeout(300);
+    my $_monitor_start_time = time();
+    my $done;
+    while ((time() - $_monitor_start_time <= $timeout) && (!$done)) {
+        sleep 30;
+        my $prov = get_required_var('PUBLIC_CLOUD_PROVIDER');
+        my $show_attr = qesap_ansible_script_output(
+            cmd => 'SAPHanaSR-showAttr',
+            provider => $prov,
+            host => $host,
+            root => 1);
+
+        my %status = ();
+        for my $line (split("\n", $show_attr)) {
+            $status{$1} = $line if ($line =~ $regular_expression);
```

Comment:
I think that this line:

```
$status{$1} = $line if ($line =~ $regular_expression);
```

and in particular `$1` only works if user provides a regexp using groups. There's not validation about it. can we may just use a list for `status`?

Looking at the code ... do we really need both $regular_expression and $text_to_find ?